dotcom-rendering/src/web/components/SecureSignupIframe.importable.tsx

```diff
@@ -0,0 +1,373 @@
+import { css } from '@emotion/react';
+import { neutral, space, until } from '@guardian/source-foundations';
+import {
+	Button,
+	InlineError,
+	InlineSuccess,
+	Link,
+	SvgReload,
+	SvgSpinner,
+} from '@guardian/source-react-components';
+import type { ReactEventHandler } from 'react';
+import { useRef, useState } from 'react';
+import ReCAPTCHA from 'react-google-recaptcha';
+import type { OphanAction } from '../browser/ophan/ophan';
+import {
+	getOphanRecordFunction,
+	submitComponentEvent,
+} from '../browser/ophan/ophan';
+
+const isServer = typeof window === 'undefined';
+
+// The Google documentation specifies that if the 'recaptcha-badge' is hidden,
+// their T+C's must be displayed instead. While this component hides the
+// badge, its parent must include the T+C along side it.
+// The T+C are not included in this componet directly to reduce layout shift
+// from the island hydrating (placeholder height for the text can't
+// be accurately predicated for every breakpoint).
+// https://developers.google.com/recaptcha/docs/faq#id-like-to-hide-the-recaptcha-badge.-what-is-allowed
+
+type Props = {
+	styles: string;
+	html: string;
+	newsletterId: string;
+	successText: string;
+};
+
+const ErrorMessageWithAdvice = ({ text }: { text?: string }) => (
+	<InlineError>
+		<span>
+			{text} Please try again or contact{' '}
+			<Link href="mailto:customer.help@theguardian.com" target="_blank">
+				customer.help@theguardian.com
+			</Link>
+		</span>
+	</InlineError>
+);
+
+const SuccessMessage = ({ text }: { text?: string }) => (
+	<InlineSuccess>
+		<span>
+			<b>Subscription Confirmed.&nbsp;</b>
+			<span>{text}</span>
+		</span>
+	</InlineSuccess>
+);
+
+const buildFormData = (
+	emailAddress: string,
+	newsletterId: string,
+	token: string,
+): FormData => {
+	const pageRef = window.location.origin + window.location.pathname;
+	const refViewId = window.guardian.ophan?.pageViewId ?? '';
+
+	const formData = new FormData();
+	formData.append('email', emailAddress);
+	formData.append('csrfToken', ''); // TO DO - PR on form handlers in frontend/identity to see how/if this is needed
+	formData.append('listName', newsletterId);
+	formData.append('ref', pageRef);
+	formData.append('refViewId', refViewId);
+	formData.append('name', '');
+	if (window.guardian.config.switches.emailSignupRecaptcha) {
+		formData.append('g-recaptcha-response', token); // TO DO - PR on form handlers - is the token verified?
+	}
+
+	return formData;
+};
+
+const postFormData = async (
+	endpoint: string,
+	formData: FormData,
+): Promise<Response> => {
+	const requestBodyStrings: string[] = [];
+
+	formData.forEach((value, key) => {
+		requestBodyStrings.push(
+			`${encodeURIComponent(key)}=${encodeURIComponent(
+				value.toString(),
+			)}`,
+		);
+	});
+
+	return fetch(endpoint, {
+		method: 'POST',
+		body: requestBodyStrings.join('&'),
```

Comment:
Can't you just pass the `formData` as-is? https://developer.mozilla.org/en-US/docs/Web/API/FormData

You might have to set the encoding type to `multipart/form-data`.

If not, maybe using a `map` instead of pushing fields to an array would keep things more functional.